r/tests/testthat/test-dplyr-funcs-datetime.R

```diff
@@ -49,50 +49,79 @@ test_df <- tibble::tibble(
 
 test_that("strptime", {
   t_string <- tibble(x = c("2018-10-07 19:04:05", NA))
-  t_stamp <- tibble(x = c(lubridate::ymd_hms("2018-10-07 19:04:05"), NA))
+  # lubridate defaults to "UTC" as timezone => t_stamp is in "UTC"
+  t_stamp_with_utc_tz <- tibble(x = c(lubridate::ymd_hms("2018-10-07 19:04:05"), NA))
+  t_stamp_with_pm_tz <- tibble(
+    x = c(lubridate::ymd_hms("2018-10-07 19:04:05", tz = "Pacific/Marquesas"), NA)
+  )
+
+  # base::strptime returns a POSIXlt (a list) => we cannot use compare_dplyr_binding
+  # => we use expect_equal for the tests below
+
+  withr::with_timezone("Pacific/Marquesas", {
+    # the default value for strptime's `tz` argument is "", which is interpreted
+    # as the current timezone. we test here if the strptime binding picks up
+    # correctly the current timezone (similarly to the base R version)
+    expect_equal(
+      t_string %>%
+        record_batch() %>%
+        mutate(
+          x = strptime(x, format = "%Y-%m-%d %H:%M:%S")
+        ) %>%
+        collect(),
+      t_stamp_with_pm_tz
+    )
+  })
+
+  # adding a timezone to a timezone-naive timestamp works
```

Comment:
```suggestion
  # adding a timezone to a timezone-naive timestamp works
  # and since our TZ when running the test is (typically) Pacific/Marquesas
  # this also tests that assigning a TZ different from the current session one
  # works as expected
```